From 8bf9a73d04c24aaaf4a65d26f75fcb11fabf49f0 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Mon, 11 Jul 2022 14:52:16 +0300
Subject: [PATCH] [SW94985]: Revert "[SW-94323] gaudi2: reset cold_rst recovery_flag in erom_recovery"

This reverts commit d0c34d36a7dbbdfe15634a64b3358a63fb8345b4.
Since the recovery flag is cleared in erom_init if the firmware update
fails, the arcboot rom will try to load ppboot instead of executing the
erom recovery. Therefore the recovery flag should be cleared only after
the successful firmware update i.e, in zephyr agent and hl-fw-loader.

Change-Id: I9b681431b30adc817a66b4750b5688f8f68654fc
---

diff --git a/app/ext_rom/erom_recovery.c b/app/ext_rom/erom_recovery.c
index 039effd..16aa192 100644
--- a/app/ext_rom/erom_recovery.c
+++ b/app/ext_rom/erom_recovery.c
@@ -87,8 +87,6 @@
 	device_init_i2c_s();
 }
 
-__weak void preboot_clear_recovery_flag(void) {}
-
 void erom_main_function(void)
 {
 	u64 ver_off = (uintptr_t)preboot_vers;
@@ -135,8 +133,6 @@
 	preboot_dev_enable_caps();
 
 	preboot_clear_wd_gpio();
-
-	preboot_clear_recovery_flag();
 
 	cpu_boot_sts_set(CPU_BOOT_STATUS_WAITING_FOR_BOOT_FIT);
 
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index d3aed0c..0a6bfd7 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1088,16 +1088,6 @@
 	clr_flr_reset();
 }
 
-#define COLD_RST_FLOPS_RECOVERY_FLAG_MASK	0x1
-void preboot_clear_recovery_flag(void)
-{
-	struct gaudi2_cold_rst_data cold_rst_data;
-
-	cold_rst_data.data = regmap_readl(mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
-	cold_rst_data.recovery_flag = 1;
-	regmap_writel(cold_rst_data.data, mmPSOC_GLOBAL_CONF_COLD_RST_FLOPS_2);
-}
-
 /*
  * As per VLSI team drain time out of 0x2000 corresponds to 16 micro seconds
  * and this value should be enough for AXI drain time out
diff --git a/include/preboot.h b/include/preboot.h
index 9ddee95..be68257 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -257,7 +257,6 @@
 
 void hl_flr_report_err(u32 err_bit);
 void preboot_security_sp_sram_init(void);
-void preboot_clear_recovery_flag(void);
 
 #endif /* __ASSEMBLY__ */
 
